std/algorithm/searching.d

```diff
@@ -638,7 +638,7 @@ Returns:
 */
 size_t count(alias pred = "a == b", Range, E)(Range haystack, E needle)
 if (isInputRange!Range && !isInfinite!Range &&
-    is(typeof(binaryFun!pred(haystack.front, needle)) : bool))
+    is(typeof(binaryFun!pred(haystack.front, needle))))
```

Comment:
Better force boolean conversion instead of removing the check entirely. Otherwise the instantiation fails at a later stage with worse error messages.

```suggestion
    is(typeof(binaryFun!pred(haystack.front, needle) ? 0 : 0)))
```
(or something else that checks for boolean conversion in a condition)

Reply:
I followed the existent convention for other methods and ranges. And I would argue that the error message is worse as is because it's not clear what the error is. Because there are overloads for `count` the actual error is `Error: no overload matches for count'`. Even if I add that conversion this error message will persist. With this approach it gives something like:
```
Error: expression `unaryFun(elem)` of type `(<range element type>)` does not have a boolean value
Error: template instance `onlineapp.count!("<user predicate>", <range type>)` error instantiating
```

Reply:
I agree, I don't think there is a need for this boolean check. As mentioned in the issue, if we look at `filter` template, e.g.: https://github.com/dlang/phobos/blob/master/std/algorithm/iteration.d#L1280-L1281 . Maybe I'm not getting some deeper edge case, but I think this is fine as is.